neo4j/neo4j-browser
# Initial Apollo Client setup and ability to add/remove/update/get Project Files

**Browser File IO Feature Branch**
- Displays Project Files specific UI (save button in Editor frame and Sidebar drawer) only when certain params are passed in via Desktop
- Add in Apollo Client allowing getting/adding/removing/updating Project Files on the file system (via Relate)
- Adding a project file will currently only be added to the root of a Project in this phase. Getting/Removing/Updating can work at any level in a project for now.

**How to test this work**
- Desktop now has latest changes on master branch. Relate has a fix for token verification (on master) but not been released or updated in Desktop.
- The method I used to test this work right now is to launch dev Browser from Desktop (enabling dev mode). Either as a graph app file url or via the dev mode url input. This will launch Browser from Desktop and allow you to get the params passed to Browser from from the dev console (`new URL(location.href).search`) and copy them.
- The params will look something like `?relateApiToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MDIxNjk1NTcsImV4cCI6MTYwMjI1NTk1N30.ktiZGi4TXzSSOa1a1WlZD7HlFU-lyVFR3-koFF89v24&relateUrl=http%3A%2F%2F127.0.0.1%3A12111&neo4jDesktopProjectId=project-6d927250-a154-4d3e-9052-a372bd269705&neo4jDesktopGraphAppRootPath=&neo4jDesktopGraphAppId=neo4j-browser-1&neo4jDesktopApiUrl=http%3A%2F%2F127.0.0.1%3A11001&neo4jDesktopGraphAppClientId=cd50b2fd9f5a8704efdf52f7e2aff663065c32b6f77e0b7b27e7f9868da9da80&neo4jDesktopInternalAppId=0bcdeecb-c3f6-4bfb-a2f8-c5e3b5e009b0` . You should be done with Desktop at this point.
- Now in order to test with latest Relate, pull latest master and run relate web (may be a port conflict if you still run Desktop, so either close or pass in a port var). Then with Browser running in your web browser, paste the params in and make sure to change `relateUrl` to point to `:3000` (Relate). It should look something like `?relateApiToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MDIxNjk1NTcsImV4cCI6MTYwMjI1NTk1N30.ktiZGi4TXzSSOa1a1WlZD7HlFU-lyVFR3-koFF89v24&relateUrl=http%3A%2F%2F127.0.0.1%3A3000&neo4jDesktopProjectId=project-6d927250-a154-4d3e-9052-a372bd269705&neo4jDesktopGraphAppRootPath=&neo4jDesktopGraphAppId=neo4j-browser-1&neo4jDesktopApiUrl=http%3A%2F%2F127.0.0.1%3A11001&neo4jDesktopGraphAppClientId=cd50b2fd9f5a8704efdf52f7e2aff663065c32b6f77e0b7b27e7f9868da9da80&neo4jDesktopInternalAppId=0bcdeecb-c3f6-4bfb-a2f8-c5e3b5e009b0`
- You should now see the Project Files specific UI and graphQL calls working correctly. If you remove all the params or specifically any/all of `relateApiToken`, `neo4jDesktopApiUrl` or `neo4jDesktopProjectId`, then then Project Files specifics will not show at all (Apollo client should not cause any problems in this case)
 - Prior UI tweaks have been added in https://github.com/neo4j-devtools/relate-by-ui/pull/7 and updating a file added in https://github.com/neo4j-devtools/relate/pull/199.

**Screenshots and Videos:**
![Screenshot 2020-10-05 at 13 58 26](https://user-images.githubusercontent.com/14161129/95082511-0b486900-0713-11eb-9484-9368ce554a3c.png)

[Video 1](https://drive.google.com/file/d/14RQzj1IjC43Xfx61DKQNK4TBOdKwAK44/view) : showing files being created, updated, removed and general UI 

[Video 2](https://drive.google.com/file/d/1C1vSDfFoTIi1Iqml0JYm2aZAJt_7WrYg/view) : network error handling

[Video 3](https://drive.google.com/file/d/1C1vSDfFoTIi1Iqml0JYm2aZAJt_7WrYg/view) :  graphQL error handling

[Video 4](https://drive.google.com/file/d/1C1vSDfFoTIi1Iqml0JYm2aZAJt_7WrYg/view) : edit file error status
